x-pack/plugins/ml/public/application/explorer/explorer_charts/explorer_charts_container.js

```diff
@@ -198,20 +198,21 @@ function ExplorerChartContainer({
 
   useEffect(
     function getMapsPluginLink() {
-      if (!series) return;
       let isCancelled = false;
-      const generateLink = async () => {
-        if (!isCancelled) {
-          try {
-            const mapsLink = await getMapsLink();
-            setMapsLink(mapsLink?.path);
-          } catch (error) {
-            console.error(error);
-            setMapsLink('');
+      if (series && getChartType(series) === CHART_TYPE.GEO_MAP) {
+        const generateLink = async () => {
+          if (!isCancelled) {
+            try {
+              const mapsLink = await getMapsLink();
+              setMapsLink(mapsLink?.path);
+            } catch (error) {
+              console.error(error);
+              setMapsLink('');
+            }
           }
```

Comment:
Check for `isCancelled` is in a wrong place. `mapsLink` is defined asynchronously, hence we need to make sure that component is still mounted before updating the component state with `setMapsLink`.
```suggestion
            try {
              const mapsLink = await getMapsLink();
               if (!isCancelled) {
              setMapsLink(mapsLink?.path);
              }
            } catch (error) {
              console.error(error);
              setMapsLink('');
            }
```

Reply:
Looks like we also then need to update the location of the isCancelled in the other `useEffect` that generates the other link for the single metric viewer 👌